src/index.ts

```diff
@@ -37,10 +41,157 @@ function gistIdFromTwiddleUrl(twiddleUrl: string): string {
+    }
+    return `${accumulator}/${current}`;
+  });
+  if (!fs.existsSync(fullPath)) {
+    fs.mkdirSync(fullPath);
+  }
+}
+
+/**
+ * Unzip and extract from a given url and save the data to a path provided.
+ * @param gistUrl
+ * @param filePath
+ * @param targetDir
+ */
+async function unzipAndExtractFromUrl(gistUrl: string, filePath: string, targetDir: string): Promise<string> {
+  const gistLink = await fetch(gistUrl).then(res => res.url);
+  const gistDataUrl = `${gistLink}/${filePath}`;
+  const newZip = new AdmZip();
+  const gistData = await fetch(gistDataUrl)
+    .then(x => x.arrayBuffer())
+    .then(x => Buffer.from(x));
+  const zip = new AdmZip(gistData);
+  const zipEntries = zip.getEntries();
+
+  zipEntries.forEach(zipEntry => {
+    const fileName = zipEntry.entryName;
+    const fileContent = zip.readAsText(fileName);
+    // Here remove the top level directory
+    const newFileName = fileName.substring(fileName.indexOf('/') + 1);
+
+    // tslint:disable-next-line: no-bitwise
+    newZip.addFile(newFileName, (fileContent as unknown) as Buffer, '', 0o644 << 16);
+  });
+
+  // Generate the project root directory.
+  generateDir(targetDir);
+  newZip.extractAllTo(targetDir);
+  // tslint:disable-next-line: no-console
+  console.log(`Files extracted to path: ${targetDir}`);
+  return targetDir;
+}
+
+/**
+ * Returns the json object for a given path
+ * @param targetDir
+ * @param fileName
+ */
+function getJsonFileData(targetDir: string, fileName: string) {
+  const file = `${targetDir}/${fileName}.json`;
+  const twiddleData = JSON.parse(fs.readFileSync(file, 'utf8'));
+  return twiddleData;
+}
+
+/**
+ * Normalize the generated twiddle to a ember-like structure.
+ * @param targetDir
+ */
+function normalizeTwiddle(targetDir: string) {
+  const appDir = `${targetDir}/app`;
+  generateDir(appDir);
+  const files = fs.readdirSync(targetDir);
+  for (let i in files) {
+    const file = files[i];
+    const dirPath = file.split('.');
+    const fileName = dirPath.splice(-2, 2);
+    const newPath = `${appDir}/${dirPath.join('/')}`;
+    const filePath = `${newPath}/${fileName.join('.')}`;
+    // If the file parts length is more than 2 then there is a directory structure expected.
+    if (file.split('.').length > 2) {
+      generateDir(`${appDir}/${dirPath.join('/')}`);
+    }
+    if (fileName.length === 2 && fs.existsSync(newPath)) {
+      try {
+        if (fs.existsSync(filePath)) {
+          continue;
+        } else {
+          fs.renameSync(`${targetDir}/${file}`, filePath);
+        }
+      } catch (err) {
+        throw err;
+      }
+    }
+  }
+}
+
+/**
+ * Augment the already generated ember-like twiddle project app into a real ember app
+ * by augmenting the real Ember app's structure on top of the twiddle app.
+ * @param twiddleData
+ * @param projectName
+ */
+async function augmentEmberApp(twiddleData: any, projectName: string) {
+  const twiddleItem = JSON.parse(JSON.stringify(twiddleData));
```

Comment:
Typically I only resort to this kind of thing if I want a _deep clone_ of an object whose shape I don't know in advance.

In this case, you could do something a little less elaborate like

```suggestion
  const twiddleItemDependencies = { ...twiddleData.dependencies }
```

to create a shallow copy